pkg/utils/zonegetter/zone_getter.go

```
@@ -201,6 +211,11 @@ func (z *ZoneGetter) ListZones(filter Filter, logger klog.Logger) ([]string, err
 // If the CR does not exist or it is not ready, ListSubnets will return only the
 // default subnet.
 func (z *ZoneGetter) ListSubnets(logger klog.Logger) []nodetopologyv1.SubnetConfig {
+	if z.mode == Legacy {
+		logger.Info("ListSubnets is being called with legacy zone getter. Ignoring error")
```

nit: "Ignoring error" -> "Returning empty list of subnets"